Automattic/woocommerce-payments
# Update save my info section

Fixes 1505-gh-Automattic/woopay

#### Changes proposed in this Pull Request

Updates save my info section at checkout:

- Applies new design to both classic and Blocks checkout.
- Change its copies.
- Update tests to match new copies and the way interactions take place.

> **Note**
> 1. The copy next to the checkbox is still under discussion.
> 2. We don't know what's the link for the Learn more text yet.
>
> Check the PRD comments (link in the issue).









#### Testing instructions

Before running the tests, make sure you have Stripe events being delivered to your local WCPay server. That way the WooPay user can be correctly created.

**Run the following tests on both classic and Blocks checkouts.**

**Test the WooPay user is created**

1. Go to the store and add a product.
2. Go to checkout and fill in the form, type an email that does not belong to a WooPay user and a valid US phone.
3. The save my info section must be visible.
4. Check the UI matches the design. There will be little distinctions between classic and Blocks, but overall they should look the same.
5. Confirm the UI looks good in the mobile version.
6. Tick the checkbox and note the phone value has been copied over the phone inner input.
7. Place the order.
8. Check the WooPay user has been created.

**Test the save my info section is NOT displayed**

1. Go to the store and add a product.
2. Go to checkout and fill in the form, type an email that belongs to a WooPay user and a valid US phone.
3. The save my info section must not be visible.
4. Place the order.
5. Confirm the WooPay user was not created.

**Test the tooltip functionality**
1. Go to the store and add a product.
2. Go to checkout and fill in the form, type an email that does not belong to a WooPay user.
3. Go to the save my info section and pass the mouse over the `i` icon, next to the WooPay logo.
4. A tooltip must show up when you hover the mouse in the icon, and it should continue visible when you move the mouse to within the tooltip.

<table>
<tr>
<th>Classic checkout</th>
<th>Blocks checkout</th>
</tr>
<tr>
<td>
<img width="332" alt="image" src="https://user-images.githubusercontent.com/10233985/208112862-a6c8c22e-0926-4ac1-a25d-f2eb7669920d.png">
</td>
<td>
<img width="492" alt="image" src="https://user-images.githubusercontent.com/10233985/208112942-994a88bb-678f-4dac-8949-82f0b049a240.png">
</td>
</tr>
</table>






-------------------


**Post merge**
